ca/certificate-authority_test.go

```diff
@@ -411,6 +415,22 @@ func TestCapitalizedLetters(t *testing.T) {
 	test.AssertDeepEquals(t, expected, parsedCert.DNSNames)
 }
 
+func TestWrongSignature(t *testing.T) {
+	ctx := setup(t)
+	defer ctx.cleanUp()
+	ctx.caConfig.MaxNames = 3
+	ca, err := NewCertificateAuthorityImpl(ctx.caConfig, ctx.fc, ctx.stats, caCertFile)
+	ca.Publisher = &mocks.Publisher{}
+	ca.PA = ctx.pa
+	ca.SA = ctx.sa
+
+	// x509.ParseCertificateRequest() does not check for invalid signatures...
+	csr, _ := x509.ParseCertificateRequest(WrongSignatureCSR)
```

Comment:
Yes, it's used, it's normal, and it doesn't call os.Exit. It throws a panic that is recovered by the testing package's code. We have some code that wraps t.Fatalf but it's very common in Go test cases. It's used when you can no longer make useful tests happen if the tested condition is true. Which is what would happen in this case.